src/persist/src/mem.rs

```diff
@@ -721,25 +727,43 @@ impl Consensus for MemConsensus {
         }
         let mut store = self.data.lock().await;
 
-        let data = store.get(key);
+        let data = match store.get(key) {
+            None => None,
+            Some(values) => values.last(),
+        };
+
         let seqno = data.as_ref().map(|data| data.seqno);
 
         if seqno != expected {
             return Ok(Err(data.cloned()));
         }
 
-        store.insert(key.to_string(), new);
+        store.entry(key.to_string()).or_insert(Vec::new()).push(new);
 
         Ok(Ok(()))
     }
 
     async fn scan(
         &self,
-        deadline: Instant,
+        _deadline: Instant,
         key: &str,
         from: Option<SeqNo>,
     ) -> Result<Vec<VersionedData>, ExternalError> {
-        unimplemented!()
+        let store = self.data.lock().await;
+        let mut ret = vec![];
+        let from = from.unwrap_or(SeqNo(0));
+
+        let values = match store.get(key) {
+            None => return Ok(ret),
+            Some(values) => values,
+        };
+
+        for value in values.iter().rev() {
```

Comment:
TODO that we could binary search to find the first one and do this all with a single memcpy instead of a bunch of pushes